Update and Fix Vagrant development setup

ivnilv opened the pull request:
Hello,

This fixes the `vagrant up` type of development described in the documentation of the project.

**USAGE:**

1. Clone phpservermon repo
2. `cd` to project root folder and run `vagrant up`
3. Open http://192.168.50.100/ in your browser and type new admin password
4. Enjoy your new dev environment !

**CHANGES**

- Updated to Ubuntu 20.04 bento vagrant box
- Updated PHP to PHP-FPM 7.4
- Updated MySQL to Percona Server 5.7
- Now runs with NGINX as web server (default vhost kept as before http://192.168.50.100/)
- **DEPRECATED:** No longer relying on puphpet code (it's outdated and no longer maintained)
- Updated `Vagrantfile` to use ansible 
- **NEW:** Automation in vagrant is now handled by **ansible** ([see provision.yaml for details](https://raw.githubusercontent.com/ivnilv/phpservermon/develop/provision.yaml))
  - A couple of shell scripts to handle "re-running" of the `vagrant up` command for faster executions ( [setup-php.sh](https://raw.githubusercontent.com/ivnilv/phpservermon/develop/setup-php.sh), [setup-mysql.sh](https://raw.githubusercontent.com/ivnilv/phpservermon/develop/setup-mysql.sh) )

**REQUIREMENTS**
- VirtualBox 6+
- Ansible on host machine (available via brew for MacOS; via apt/yum/pacman/zypper for Linux)

ivnilv committed:
replaced puphpet with ansible configuration and updated Vagrant images

ivnilv committed:
added back puphpet files, cleaning up should be handled in a separate task

ivnilv committed:
restart nginx at the end to apply changes

ivnilv commented:
@TimZ99 mind taking a look ? :)

ivnilv commented:
Hey @TimZ99 , I was really excited as this is one of my first public attempts for a PR to an OSS project. Hope you have some constructive feedback here and would love to fix stuff that you and the community would see not fitting right :)
Thanks 🤞

TimZ99 committed:
Removed trailing spaces

TimZ99 commented on dev/phpservermon-default:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    location = /robots.txt {
+        allow all;
```
As this is for a dev env I think we don't want crawlers.

ivnilv replied:
True, but in reality this would be a VM with just internal access so essentially no crawler should ever reach it. 

As it doesn't serve any actual purpose I'll indeed remove it.

TimZ99 commented on dev/phpservermon-default:
```diff
@@ -0,0 +1,84 @@
+}
+
+server {
+    listen 80;
```
Would it support https?

ivnilv replied:
I can add HTTPS support, as right now it's not there.

The question is how would it be handled, an easy and dirty fix I see is go issue a self-signed cert for the VM IP address (192.168.50.100) however we must be sure that the IP of the VM doesnt change, otherwise the certificate won't work.

TimZ99 commented on provision.yaml:
```diff
@@ -0,0 +1,85 @@
+  - name: ensure percona-server-57 is installed
+    apt: name=percona-server-server-5.7 state=latest
+
+    #  - name: Create a new database with name 'psm'
```
Should this be removed?

ivnilv replied:
This is the part that installs the percona mysql database inside the VM, so it's there on purpose.

TimZ99 commented on setup-mysql.sh:
```diff
@@ -0,0 +1,8 @@
+# Create user
+mysql -e "CREATE USER 'psm'@'localhost' IDENTIFIED BY 'psm-dev-password';"
+mysql -e "GRANT ALL PRIVILEGES ON psm.* TO 'psm'@'localhost';"
+mysql -e "FLUSH PRIVILEGES;";
```
last ; not needed right?

ivnilv replied:
Oh yea, I'll clean that up :)

TimZ99 commented on provision.yaml:
```diff
@@ -0,0 +1,85 @@
+     name: '*'
+     state: latest
+
+     #  - name: Install base packages
```
Could this be removed?

ivnilv replied:
I think this is good to stay. 

It ensures that the VM runs with the latest software, taking care of security/feature/bug fixes.

TimZ99 replied:
Sorry, I meant the part that is commented out.

TimZ99 commented on Vagrantfile:
```diff
@@ -1,264 +1,27 @@
+  vb.cpus = "2"
+  end
+
+  #config.vm.provider :libvirt do |v|
```
Could this be removed

ivnilv replied:
*redacted*

ivnilv replied:
removed the commented stuff

TimZ99 reviewed (commented):
Nice clean and short code. Missing an updated documentation file.

TimZ99 commented:
Hi @ivnilv, sorry for the wait. I hope to see more pull requests from you! Don't let my slow reaction take away the excitement.

ivnilv commented:
Hi @TimZ99 

Thanks for your feedback ! Really appreciate it :)

I've added some comments and will try to push the changes as soon as possible to my dev branch.

I would love to hear your thoughts on the HTTPS/SSL support for the dev environment and if it should be added.

ivnilv committed:
Merge branch 'phpservermon:develop' into develop

ivnilv committed:
fixed: removed extra column at setup-mysql.sh script and changed access to robots.txt to deny all

ivnilv committed:
added missing space

ivnilv committed:
cleanup provision.yaml

ivnilv committed:
cleanup Vagrantfile

ivnilv committed:
updated Vagrant development documentation

ivnilv commented:
@TimZ99 I believe I've covered the stuff we discussed above. Would you mind taking another look at the MR :)

ivnilv commented:
@TimZ99  ping ? :)

TimZ99 committed:
Removed PuPHPet

- PuPHPet isn't used anymore in the current workflow.
- PuPHPet hasn't been updated sinds 2018.
- Misconfig for nginx which could lead to crlf injection.

TimZ99 commented:
CHANGES

- Updated to Ubuntu 20.04 bento vagrant box
- Updated PHP to PHP-FPM 7.4
- Updated MySQL to Percona Server 5.7
- Now runs with NGINX as web server (default vhost kept as before http://192.168.50.100/)
- DEPRECATED: No longer relying on puphpet code (it's outdated and no longer maintained)
- Updated Vagrantfile to use ansible
- NEW: Automation in vagrant is now handled by ansible (see provision.yaml for details)
- A couple of shell scripts to handle "re-running" of the vagrant up command for faster executions ( setup-php.sh, setup-mysql.sh )

TimZ99 commented:
It took some pinging, but it's finally merged 😄 .
Thanks a lot!
